lib/rpc/GlobalRpcProviders.ts

```diff
@@ -2,87 +2,89 @@ import { ChainId } from '@uniswap/sdk-core'
 import { SingleJsonRpcProvider } from './SingleJsonRpcProvider'
 import { UniJsonRpcProvider } from './UniJsonRpcProvider'
 import Logger from 'bunyan'
-import { SUPPORTED_CHAINS } from '@uniswap/smart-order-router'
 import {
   DEFAULT_SINGLE_PROVIDER_CONFIG,
   DEFAULT_UNI_PROVIDER_CONFIG,
   SingleJsonRpcProviderConfig,
   UniJsonRpcProviderConfig,
 } from './config'
+import { ChainConfig, ProdConfig } from './ProdConfig'
+import { chainIdToNetworkName } from './utils'
 
 export class GlobalRpcProviders {
-  private static readonly PROVIDER_RPC_URL_RANKING: Map<ChainId, number[] | undefined> = new Map([
-    [ChainId.AVALANCHE, undefined],
-  ])
-
-  private static readonly PROVIDER_RPC_URL_WEIGHTS: Map<ChainId, number[] | undefined> = new Map([
-    [ChainId.AVALANCHE, undefined],
-  ])
-
   private static SINGLE_RPC_PROVIDERS: Map<ChainId, SingleJsonRpcProvider[]> | null = null
 
   private static UNI_RPC_PROVIDERS: Map<ChainId, UniJsonRpcProvider> | null = null
 
-  private static initGlobalSingleRpcProviders(log: Logger, config: SingleJsonRpcProviderConfig) {
-    // Only Avalanche is supported for now.
-    const infuraAvalancheUrl = process.env[`WEB3_RPC_${ChainId.AVALANCHE.toString()}`]!
-    if (infuraAvalancheUrl === undefined) {
-      throw new Error(
-        `Infura Avalanche URL isn't provided by environment variable WEB3_RPC_${ChainId.AVALANCHE.toString()}`
-      )
+  private static getProdConfig(): ProdConfig {
+    const prodConfigStr = process.env['UNI_RPC_PROVIDER_PROD_CONFIG']!
+    if (prodConfigStr === undefined) {
+      throw new Error('Environment variable UNI_RPC_PROVIDER_PROD_CONFIG is missing!')
+    }
+    // Notice: No input validation! You need to be careful about the env var's format!
```

Comment:
this is very risky... shall we default to something then? if someone pushes the wrong enviornment then we can completely break it right?

Reply:
> Thanks for adding this @rollingtumbling - one small point, historically we've ran either https://joi.dev/api/?v=17.12.0 for schema validation in the Routing API. Very similar to io-ts, just without the functional bend to it. Mind using that here as well? We've got a handful of examples here:
> 
> https://github.com/Uniswap/routing-api/blob/147f2204d6ee7ab78fe9a44fb48114707476e554/lib/handlers/quote/schema/quote-schema.ts#L22

Now use Joi instead. Please take another look!